cmake/modules/FindDEAL_II_TRILINOS.cmake

```diff
@@ -66,129 +62,135 @@ if(DEFINED Trilinos_VERSION)
-if(TARGET Kokkos::kokkos)
-  get_property(KOKKOS_COMPILE_FLAGS_FULL TARGET Kokkos::kokkos PROPERTY INTERFACE_COMPILE_OPTIONS)
-  string(REGEX REPLACE "\\$<\\$<COMPILE_LANGUAGE:CXX>:([^>]*)>" "\\1" KOKKOS_COMPILE_FLAGS "${KOKKOS_COMPILE_FLAGS_FULL}")
-  string(REPLACE ";" " " KOKKOS_COMPILE_FLAGS "${KOKKOS_COMPILE_FLAGS}")
+  remove_duplicates(Trilinos_TPL_INCLUDE_DIRS)
+
+  if(TARGET Kokkos::kokkos)
+    get_property(KOKKOS_COMPILE_FLAGS_FULL TARGET Kokkos::kokkos PROPERTY INTERFACE_COMPILE_OPTIONS)
+    string(REGEX REPLACE "\\$<\\$<COMPILE_LANGUAGE:CXX>:([^>]*)>" "\\1" KOKKOS_COMPILE_FLAGS "${KOKKOS_COMPILE_FLAGS_FULL}")
+    string(REPLACE ";" " " KOKKOS_COMPILE_FLAGS "${KOKKOS_COMPILE_FLAGS}")
+
+    #
+    # Extract missing openmp linker options from the
+    # "INTERFACE_LINK_LIBRARIES" property of the Kokkos::kokkos target:
+    #
+    set(_kokkos_openmp_flags)
+    get_property(_libraries TARGET Kokkos::kokkos PROPERTY INTERFACE_LINK_LIBRARIES)
+    foreach(_entry ${_libraries})
+      if(${_entry} MATCHES "^-f?openmp")
+        add_flags(_kokkos_openmp_flags "${_entry}")
+      endif()
+    endforeach()
+
+    # Some Kokkos versions included in Trilinos before 13.2.0 add "-x cuda" when 
+    # using clang++ as compiler even if Kokkos has not been configured with Cuda
+    # support. Simply strip that flag from what we are using in that case.
+    if(NOT Kokkos_ENABLE_CUDA)
+      string(REPLACE "-x cuda" "" KOKKOS_COMPILE_FLAGS "${KOKKOS_COMPILE_FLAGS}")
+    endif()
+  endif()
 
   #
-  # Extract missing openmp linker options from the
-  # "INTERFACE_LINK_LIBRARIES" property of the Kokkos::kokkos target:
+  # We'd like to have the full library names but the Trilinos package only
+  # exports a list with short names...
+  # So we check again for every lib and store the full path:
   #
-  set(_kokkos_openmp_flags)
-  get_property(_libraries TARGET Kokkos::kokkos PROPERTY INTERFACE_LINK_LIBRARIES)
-  foreach(_entry ${_libraries})
-    if(${_entry} MATCHES "^-f?openmp")
-      add_flags(_kokkos_openmp_flags "${_entry}")
-    endif()
+  set(_libraries "")
+  foreach(_library ${Trilinos_LIBRARIES})
+    list(APPEND _libraries TRILINOS_LIBRARY_${_library})
+    deal_ii_find_library(TRILINOS_LIBRARY_${_library}
+      NAMES ${_library}
+      HINTS ${Trilinos_LIBRARY_DIRS}
+      NO_DEFAULT_PATH
+      NO_CMAKE_ENVIRONMENT_PATH
+      NO_CMAKE_PATH
+      NO_SYSTEM_ENVIRONMENT_PATH
+      NO_CMAKE_SYSTEM_PATH
+      NO_CMAKE_FIND_ROOT_PATH
+      )
   endforeach()
 
-  # Some Kokkos versions included in Trilinos before 13.2.0 add "-x cuda" when 
-  # using clang++ as compiler even if Kokkos has not been configured with Cuda
-  # support. Simply strip that flag from what we are using in that case.
-  if(NOT Kokkos_ENABLE_CUDA)
-    string(REPLACE "-x cuda" "" KOKKOS_COMPILE_FLAGS "${KOKKOS_COMPILE_FLAGS}")
-  endif()
+  process_feature(TRILINOS
+    LIBRARIES
+      REQUIRED ${_libraries}
+      OPTIONAL Trilinos_TPL_LIBRARIES MPI_CXX_LIBRARIES
```

Comment:
Can we change this to simply read
```
process_feature(TRILINOS
  TARGETS REQUIRED ${_target}
  ...
  )
```
for Trilinos 14 onwards? (I would like the logic to simply be #15156 for Trilinos 14 onwards)

Reply:
This for Trilinos < 14, for Trilinos 14, we have
```CMake
set(_target Trilinos::all_libs)
  process_feature(TRILINOS
    TARGETS REQUIRED _target
    CLEAR
      EPETRA_CONFIG_H TRILINOS_CXX_SUPPORTS_SACADO_COMPLEX_RAD
    )
```

Reply:
Excellent.